tests/components/homekit_controller/test_media_player.py

```diff
@@ -202,3 +229,19 @@ def create_tv_service_with_target_media_state(accessory):
     )
     assert helper.characteristics[REMOTE_KEY].value is None
     assert helper.characteristics[TARGET_MEDIA_STATE].value is None
+
+
+async def test_tv_set_source(hass, utcnow):
+    """Test that we can set the input source of a HomeKit TV."""
+    helper = await setup_test_component(hass, create_tv_service)
+
+    await hass.services.async_call(
+        "media_player",
+        "select_source",
+        {"entity_id": "media_player.testdevice", "source": "HDMI 2"},
+        blocking=True,
+    )
+    assert helper.characteristics[ACTIVE_IDENTIFIER].value == 2
+
+    state = await helper.poll_and_get_state()
+    assert state.attributes["source"] == "HDMI 2"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5141152)